src/util/dataAccess.cpp

```diff
@@ -165,21 +165,114 @@ ndsize_t positionToIndex(double position, const string &unit, const RangeDimensi
 }
 
 
+void getMaxExtent(const Dimension &dim, ndsize_t max_index, double &pos, double &ext) {
+    DimensionType dt = dim.dimensionType();
+    if (dt == DimensionType::Sample) {
+        SampledDimension sd = dim.asSampledDimension();
+        pos = sd.positionAt(0);
+        ext = sd.positionAt(max_index);
+    } else if (dt == DimensionType::Range) {
+        RangeDimension rd = dim.asRangeDimension();
+        pos = rd.tickAt(0);
+        ext = rd.tickAt(max_index);
+    } else if (dt == DimensionType::Set) {
+        SetDimension sd = dim.asSetDimension();
+        pos = 0.0;
+        if (max_index > pow(FLT_RADIX, std::numeric_limits<double>::digits)) {
+            throw nix::OutOfBounds("dataAccess::fillPositionsExtents: shape cannot be cast to double without loss of precision. Please open an issue on github!");
+        }
+        ext = static_cast<double>(max_index);
+    }
+}
+
+
+vector<pair<double, double>> maximumExtents(const DataArray &array) {
+    vector<Dimension> dimensions = array.dimensions();
+    vector<pair<double, double>> max_extents;
+    for (size_t i = 0; i < dimensions.size(); ++i) {
+        double pos, ext;
+        getMaxExtent(dimensions[i], array.dataExtent()[i]-1, pos, ext);
+        max_extents.emplace_back(pos, ext);
+    }
+    return max_extents;
+}
+
+
+string getDimensionUnit(const Dimension &dim) {
+    DimensionType dim_type = dim.dimensionType();
+    string unit;
+    switch (dim_type) {
+    case DimensionType::Set:
+        unit = "none";
+        break;
+    case DimensionType::DataFrame: {
+        DataFrameDimension df_dim = dim.asDataFrameDimension();
+        unit = "none";
+        if (df_dim.columnIndex()) {
+            unit = df_dim.unit();
+        }
+        if (unit.empty()) {
+            unit = "none";
+        }
+        break;
+    }
+    case DimensionType::Sample: {
+        SampledDimension sd = dim.asSampledDimension();
+        unit = sd.unit() ? *sd.unit() : "none";
+        break;
+    }
+    case DimensionType::Range: {
+        RangeDimension rd = dim.asRangeDimension();
+        unit = rd.unit() ? *rd.unit() : "none";
```

Comment:
`rd.unit.value_or("none")`

Reply:
Citing boost: ``Effects: Equivalent to if (*this) return **this; else return std::forward<U>(v);``

but easier to read, true

Reply:
I was not implying your code was wrong ;)

Reply:
got that, and I agree that you suggestion is easier to read but since I came to like the ternary operator...
When you got a hammer, everything looks like a nail ...